### ISSUE TITLE
Predict fails on slices of training data.

### ISSUE THREAD

**Issue by paulsbrookes** [uber/orbit#761]
**Describe the bug**
If we try to produce a forecast on a slice of training data which does not start at the beginning of said training data, then the forecaster fails and raises an error.

**To Reproduce**
We can adapt the example code from the front page of the repo:

```
from orbit.utils.dataset import load_iclaims
from orbit.models import DLT
from orbit.diagnostics.plot import plot_predicted_data

# log-transformed data
df = load_iclaims()
# train-test split
test_size = 52
train_df = df[:-test_size]
test_df = df[-test_size:]

dlt = DLT(
  response_col='claims', date_col='week',
  regressor_col=['trend.unemploy', 'trend.filling', 'trend.job'],
  seasonality=52,
)
dlt.fit(df=train_df)

# outcomes data frame
predicted_df = dlt.predict(df=train_df[1:])
```

This produces the following error:

```
RuntimeError                              Traceback (most recent call last)
/tmp/ipykernel_93980/4190327741.py in <cell line: 20>()
     18 
     19 # outcomes data frame
---> 20 predicted_df = dlt.predict(df=train_df.iloc[1:])
     21 
     22 plot_predicted_data(

~/anaconda3/envs/orbit/lib/python3.8/site-packages/orbit/forecaster/full_bayes.py in predict(self, df, decompose, store_prediction_array, seed, **kwargs)
     98             ) if self._n_bootstrap_draws > 1 else self._posterior_samples
     99 
--> 100             predicted_dict = self._model.predict(
    101                 posterior_estimates=posterior_samples,
    102                 df=df,

~/anaconda3/envs/orbit/lib/python3.8/site-packages/orbit/template/dlt.py in predict(self, posterior_estimates, df, training_meta, prediction_meta, include_error, **kwargs)
    650 
    651         # sum components
--> 652         pred_array = trend_component + seasonal_component + regression
    653 
    654         pred_array = pred_array.numpy()

RuntimeError: The size of tensor a (389) must match the size of tensor b (390) at non-singleton dimension 1
```
However if we change the final line to start at the beginning of the training data then we have no problem. For example:

```
predicted_df = dlt.predict(df=train_df.iloc[[0, 1, 10]])
```

**Expected behavior**
We should be able to produce forecasts for any chosen selection of rows in train. E.g. we should be able to run the following forecast:
```
predicted_df = dlt.predict(df=train_df.iloc[[10, 15, 17]])
```
Ideally we would able to make predictions for selected rows of train and test data at the same time e.g.

```
predicted_df = dlt.predict(df=df.iloc[[100, 200, 300, 400]])
```

**Environment (please complete the following information):**
 - OS: Ubuntu 20.04.4 LTS
 - Python Version: 3.8
 - Versions of Major Dependencies: `pandas=='1.4.2'`, `scikit-learn==0.22`, `numpy==1.21.6` and `orbit==1.1.1`.

**Comment by Abutler101** [comment]
Using the test: 
```python
@pytest.mark.parametrize(
    "slice_range", [
        (0, None),
        (1, None),
        (0, -1),
        (36, 72)
    ],
    ids=[
        "all-train",
        "all-except-first-elem",
        "all-except-last-elem",
        "arbitrary-subset",
    ],
)
def test_dlt_predict_on_train_slice(make_weekly_data, slice_range: Tuple[int, int]):
    # log-transformed data
    df = load_iclaims()
    # train-test split
    test_size = 52
    train_df = df[:-test_size]
    test_df = df[-test_size:]

    dlt = DLT(
        response_col="claims",
        date_col="week",
        regressor_col=["trend.unemploy", "trend.filling", "trend.job"],
        seasonality=52,
    )
    dlt.fit(df=train_df)
    predicted_df = dlt.predict(df=train_df[slice_range[0]:slice_range[1]])
    print(f"Sample of Prediction:\n{predicted_df.head(5)}")
```
You can see that moving from v1.1.1 -> v1.1.2 resolves this issue

**Comment by paulsbrookes** [comment]
The issue is solved in v1.1.2 so I'm closing it now.
